### PR TITLE
 HARU-43-aws-client-uploads

### PR DESCRIPTION
**Pull request by patrickchin**

### CODE REVIEW

File: app/project/[id]/task/[taskid]/components/task-comments.tsx [hunk 1]
```diff
@@ -291,16 +307,17 @@ function UploadAttachment({
...
 
-    // server action arguments can only be primatives or FormData
-    const data = new FormData();
-    data.set("file", file);
-    const newFile = await addTaskFile(taskId, data);
-    // as DesignFile might be a sign of bad typing ... TODO improve
-    if (newFile) setCurrentAttachments((l) => [...l, newFile as DesignFile]);
+    const selectedFiles = Array.from(targetFiles);
+
+    for (let i = 0; i < selectedFiles.length; i++) {
```
Comment (patrickchin) [inline app/project/[id]/task/[taskid]/components/task-comments.tsx:314]: I think this for loop can be replaced by a `for (const file of targetFiles)` type loop

File: app/project/[id]/task/[taskid]/components/task-files.tsx [hunk 2]
```diff
@@ -45,15 +53,16 @@ export default function TaskFiles({ taskId }: { taskId: number }) {
...
-    const file = targetFiles.item(0);
-    assert(file);
+    const selectedFiles = Array.from(targetFiles);
 
-    // server action arguments can only be primatives or FormData
-    const data = new FormData();
-    data.set("file", file);
-    const newFiles = await addTaskFileReturnAll(taskId, data);
-    mutate(newFiles, { revalidate: false });
+    for (let i = 0; i < selectedFiles.length; i++) {
```
Comment (patrickchin) [inline app/project/[id]/task/[taskid]/components/task-files.tsx:58]: same here

File: lib/actions.ts [hunk 3]
```diff
@@ -136,8 +137,11 @@ export async function submitProjectForm2(formData: FormData) {
     });
   });
 
+  console.log("Results", results, "Files:", files);
```
Comment (patrickchin) [inline lib/actions.ts:140]: can remove

File: lib/s3.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    ],
+    Fields: {
+      acl: "public-read",
+      "Content-Type": contentType,
+    },
+    Expires: 600, // Presigned URL expiration in seconds
+  });
+}
+
+export async function deleteFileFromS3(key: string) {
```
Comment (patrickchin) [inline lib/s3.ts:22]: can you verify that this is called? I don't see either of the two logs messages below show up

File: app/api/upload/[type]/route.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  let uniqueKey = "";
+
+  if (type === "avatar") {
+    const { filename } = params;
+    const userId = Number(session.user.id);
+    uniqueKey = `user/${userId}/${filename}`;
+  } else if (type === "project-files") {
+    const { filename, projectId, specId } = params;
+    uniqueKey = `project/${projectId}/task/${specId}/${filename}`;
```
Comment (patrickchin) [inline app/api/upload/[type]/route.ts:25]: these values should be validated to makesure they aren't undefined
but also the project and task spec should exist too

as is could allow for someone to upload a file to a project that doesn't exist yet, and when that project with that id is created that file would already exist for that task, could be like a fake legal document or a different design etc. not ideal

File: app/project/[id]/task/[taskid]/components/task-comments.tsx (hunk 1 above)
Comment (emilielydiat) [reply app/project/[id]/task/[taskid]/components/task-comments.tsx:314]: Sure thing - sorted!

File: app/project/[id]/task/[taskid]/components/task-files.tsx (hunk 2 above)
Comment (emilielydiat) [reply app/project/[id]/task/[taskid]/components/task-files.tsx:58]: Done

File: lib/actions.ts (hunk 3 above)
Comment (emilielydiat) [reply lib/actions.ts:140]: Noted!

File: lib/s3.ts (hunk 4 above)
Comment (emilielydiat) [reply lib/s3.ts:22]: I'm seeing it being called: 

Screenshot 2024-06-26 at 23 36 54

Are you able to check in the S3 Bucket (I don't have access)?

File: app/settings/page.tsx [hunk 6]
```diff
@@ -95,22 +80,13 @@ function SettingsPage() {
               {errorMessage}
             </p>
           )}
-          <div className="flex gap-2">
```
Comment (patrickchin) [inline app/settings/page.tsx:98]: add this back!

**Pull request merged**